app/controllers/samples_controller.rb

```diff
@@ -1111,6 +1111,34 @@ def upload_heartbeat
     render json: {}, status: :ok
   end
 
+  def coverage_viz_summary
+    pr = select_pipeline_run(@sample, params)
+    coverage_viz_summary_s3_path = pr.coverage_viz_summary_s3_path
+
+    if coverage_viz_summary_s3_path
+      @coverage_viz_summary = get_s3_file(coverage_viz_summary_s3_path)
+      render json: @coverage_viz_summary
+    else
+      render json: {
+        error: "coverage viz summary file does not exist for this sample"
+      }
+    end
+  end
+
```

Comment:
you may still want some sort of rescue => json error for these two in case a fetching step fails